sdk/core/azure-core/inc/uuid.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cstdlib> // for abort
+#include <new> // for placement new
+#include <string>
+#include <type_traits>
+#include <utility> // for swap and move
+
+namespace Azure { namespace Core {
+
+  class UUID {
+
+  private:
+    static const size_t UUIDSize = 16;
+
+    unsigned char m_uuid[UUIDSize];
+    // The UUID reserved variants.
+    static const unsigned char ReservedNCS = 0x80;
+    static const unsigned char ReservedRFC4122 = 0x40;
+    static const unsigned char ReservedMicrosoft = 0x20;
+    static const unsigned char ReservedFuture = 0x00;
+
+  public:
+    UUID()
+    {
+      std::random_device rd;
+      std::mt19937 gen(rd());
+      for (int i = 0; i < UUIDSize; i++)
+        m_uuid[i] = (unsigned char)gen();
```

Comment:
Conclusion from Jun. 10 meeting:

* Use C++ style casts
* Avoid C casts.

reference to the recording at around 52 min.